tradefed/__init__.py

```diff
@@ -5,30 +5,39 @@
 import xmlrpc
 import yaml
 import xml.etree.ElementTree as ET
-#from celery import chord as celery_chord
+from celery import chord as celery_chord
 from io import BytesIO
 from django.conf import settings
 from django.core.files.base import ContentFile
 from django.db import transaction
 from squad.plugins import Plugin as BasePlugin
 from urllib.parse import urljoin
-#from squad.celery import app as celery
+from squad.celery import app as celery
 from squad.core.utils import join_name
-from squad.core.models import Suite, SuiteMetadata, Test, KnownIssue, Status, TestRun, ProjectStatus
+from squad.core.models import Suite, SuiteMetadata, Test, KnownIssue, Status, TestRun, ProjectStatus, PluginScratch
 from squad.core.tasks import get_suite
 
 
 logger = logging.getLogger()
 
 
-#@celery.task
+@celery.task
 def update_build_status(results_list, testrun_id):
     testrun = TestRun.objects.get(pk=testrun_id)
     ProjectStatus.create_or_update(testrun.build)
 
 
-#@celery.task
-def create_testcase_tests(test_case_string, atomic_test_suite_name, testrun_id, suite_id):
+@celery.task
+def create_testcase_tests(test_case_string_storage_id, atomic_test_suite_name, testrun_id, suite_id):
+    test_case_string = None
+    scratch_object = None
+    try:
+        scratch_object = PluginScratch.objects.get(pk=test_case_string_storage_id)
+        test_case_string = scratch_object.storage
```

Comment:
Don't you need `decode` here?

Reply:
No. It's weird how that works. ET.tostring() returns bytes object. So to save to db properly I need to call decode("utf-8") on what ET.tostring() returns (line 244). This means that database stores string which can be directly used in ET.fromstring().

Reply:
That's good then!